File: tests/fixtures/ksp-r41-z2-p2/sol-8-dusan.py
T = int(input())
for t in range(T):
    sifra = input()
    kluc = {}

    ktore = 0
    for znak in sifra:
        if znak not in kluc:
            if ktore == 0:
                kluc[znak] = 1
            elif ktore == 1:
                kluc[znak] = 0
            else:
                kluc[znak] = ktore
            ktore += 1

    odpoved = 0
    dlzka_sifry = len(sifra)
    zaklad = max(2, len(kluc))
    for i in range(dlzka_sifry):
        odpoved += kluc[sifra[i]]*zaklad**(dlzka_sifry-i-1)
    print(odpoved)

File: tests/fixtures/ksp-r41-z2-p2/sol-8-dusan.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int T;
    cin >> T;

    while (T--) {
        string sifra;
        cin >> sifra;

        int kolkate = 0;

        map<char, int> kluc;
        for (auto ch : sifra) {
            if (kluc.count(ch) == 0) {
                if (kolkate == 0)
                    kluc[ch] = 1;
                else if (kolkate == 1)
                    kluc[ch] = 0;
                else
                    kluc[ch] = kolkate;
                kolkate++;
            }
        }

        ll zaklad = max((ll)2, (ll)kluc.size());
        ll dlzka_sifry = sifra.length();
        vector<ll> pows = {1};
        for (int e = 1; e < dlzka_sifry; ++e)
            pows.push_back(pows.back() * zaklad);

        ll odpoved = 0;
        for (int i = 0; i < dlzka_sifry; i++)
            odpoved += kluc[sifra[i]] * pows[dlzka_sifry - i - 1];
        cout << odpoved << "\n";
    }
}